pg_analytics/src/hooks/executor.rs

```diff
@@ -68,5 +78,50 @@ fn create_logical_plan(query_desc: PgBox<pg_sys::QueryDesc>) -> Result<LogicalPl
     let context_provider = ParadeContextProvider::new()?;
     let sql_to_rel = SqlToRel::new(&context_provider);
 
-    Ok(sql_to_rel.statement_to_plan(statement.clone())?)
+    let mut logical_plan: Option<LogicalPlan> = None;
+    // If functions are undefined, then try to find and register the function and then try to get the plan again
+    while true {
+        let re = Regex::new(r"Invalid function '(.+)'").unwrap();
+        match sql_to_rel.statement_to_plan(statement.clone()) {
+            Ok(plan) => {
+                logical_plan = Some(plan);
+                break;
+            },
+            Err(err) => match err {
+                DataFusionError::Plan(err_string) => {
```

Comment:
`DataFusionError::Plan` might get thrown on errors not related to missing UDFs

Reply:
So this regex actually pushes up all `DataFusionError::Plan`s that don’t match the “Invalid function” regex! I’ll add a clarifying comment though.